src/auro-datepicker.js

```diff
@@ -352,10 +325,56 @@ class AuroDatePicker extends LitElement {
     if (!this.dropdown.hasAttribute('aria-expanded')) {
       this.dropdown.setAttribute('aria-expanded', this.dropdown.isPopoverVisible);
     }
+  }
+
+  /**
+   * Binds all behavior needed to the input after rendering.
+   * @private
+   * @returns {void}
+   */
+  configureInput() {
+    this.triggerInput = this.dropdown.querySelector('[slot="trigger"');
+    this.input = this.dropdown.querySelector('auro-input');
+
+    this.input.addEventListener('auroInput-ready', () => {
+      this.auroInputReady = true;
+    });
+
+    this.input.addEventListener('blur', () => {
+      this.handleRequired();
+    });
+
+    this.input.addEventListener('input', () => {
+      this.handleInputValueChange();
+    });
+
+    // auto-show bib when manually entering a date
+    this.input.addEventListener('keypress', (evt) => {
+      if (evt.key.length === 1) {
+        this.dropdown.show();
+      }
+    });
```

Comment:
Food for thought. While it is convenient to put these event listeners here, you _could_ reference these in the HTML template. E.g. 

```
@input="this.handleInputValueChange();"
```

IMHO, this removes some of the mystery when looking at the HTML template and understanding it's expectations. 

No change is needed, but sharing perspective.

Reply:
I agree, would be good to do a cleanup on this `addEventListener` calls across select, combobox, datepicker.

Reply:
No action item required, but food for thought.